qiskit/result/sampled_expval.py

```diff
@@ -62,7 +66,16 @@ def sampled_expectation_value(dist, oper):
         oper_strs = oper.paulis.to_labels()
         coeffs = np.asarray(oper.coeffs)
     elif isinstance(oper, SparseObservable):
-        return sampled_expval_sparse_observable(oper, dist)
+        if variance:
+            if len(oper) != 1:
+                raise ValueError(
+                    "Variance calculation only supported for single Pauli string operators."
+                )
+            expectation = sampled_expval_sparse_observable(oper, dist)
+            var_stat = 1 - (expectation) ** 2
```

Comment:
Hi Kalyan, thank you so much for your contribution. This is definitely a great feature to add, and just the right implementation to do so. I had just a couple of clarifications however based on my interpretation of the issue and its solution. 

Over here, I noticed that the math may be incorrect in the case of single qubit pauli string operators with a coefficient. This snippet may become:

```suggestion
            var_stat = (coeffs[0])**2 - (expectation)**2
```

since the square of the coefficients still remain while our pauli becomes identity in the squared expectation. This may enable our logic to work also for cases with 
`obs = coefficient**X`

Reply:
Thanks @aaryav-3 for the review. When I wrote this, I had only the Pauli strings in mind (without the coefficients). But I feel your suggestion makes sense. However, the coefficients could be complex numbers, and we have to extract only the real element.

Reply:
Hi @kaldag , thank you for your updates and accommodating the changes. You are right, in the new implementation of `var_stat`, extracting the real part of the complex number, I think will be a safe bet. Considering that all single pauli states should have real coefficients, since paulis on their own form a complete real+imaginary basis for 2x2 matrices. Perhaps we can do some exception handling to check and raise if the coefficient in the input pauli is real or not? something along the lines of "Variance is intractable for paulis with complex coefficients"?